client/clientd/src/lib.rs

```diff
@@ -1,18 +1,31 @@
 use anyhow::Result;
+use axum::response::{IntoResponse, Response};
 use bitcoin::hashes::hex::ToHex;
 use bitcoin::Transaction;
 use minimint_api::{Amount, OutPoint, TransactionId};
 use minimint_core::modules::mint::tiered::coins::Coins;
 use minimint_core::modules::wallet::txoproof::TxOutProof;
 use mint_client::mint::{CoinFinalizationData, SpendableCoin};
+use mint_client::ClientError;
+use reqwest::StatusCode;
 use serde::{Deserialize, Serialize};
+use serde_json::json;
+use thiserror::Error;
 
-#[derive(Deserialize, Serialize)]
-pub enum RpcResult {
-    #[serde(rename = "success")]
-    Success(serde_json::Value),
-    #[serde(rename = "failure")]
-    Failure(serde_json::Value),
+#[derive(Error, Debug)]
+pub enum ClientdError {
+    #[error("Client error: {0}")]
+    ClientError(#[from] ClientError),
+}
+
+impl IntoResponse for ClientdError {
+    fn into_response(self) -> Response {
+        (
+            axum::http::StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
I think we should be more specific/correct here: an internal server error means that the programmer did something wrong (afaik), but some of the errors caught here can be triggered by wrong user input.

Reply:
It is also weird  that we mix `application/json` and `application/text` which is [inconsistent](https://github.com/fedimint/fedimint/pull/384/files#diff-f6b964201dc07a8c69ecdf7561cce91fe4eff7c46e1a059c32404f67acc693a0R129-R133). I guess if clientd didn't crash (which it should never) then you should always get some json in response. I'll try to find a solution and add a new commit to this PR.

Reply:
I think I'll just go with this format:
success 200
```
{
    "data": {...}
}
```
failure 400 if its a client error 500 if its something like reading db, initializing rng, sending something over channel etc goes wrong
```
{
    "error": {...}
}
```